cmd/jujud/agent/engine/config.go

```diff
@@ -29,6 +31,87 @@ func DependencyEngineConfig() dependency.EngineConfig {
 		BackoffResetTime: 1 * time.Minute,
 		MaxDelay:         2 * time.Minute,
 		Clock:            clock.WallClock,
+		Metrics:          metrics,
 		Logger:           loggo.GetLogger("juju.worker.dependency"),
 	}
 }
+
+const (
+	dependencyMetricsNamespace   = "juju"
+	dependencySubsystemNamespace = "dependency_engine"
+)
+
+const (
+	// metricsLabelWorkerName defines a constant for the MetricsWorkerStartNames
+	// Labels.
+	metricsLabelWorkerName = "worker"
+
+	// metricsLabelModelUUID defines a constant for the MetricsWorkerStartNames
+	// Labels.
+	metricsLabelModelUUID = "model_uuid"
+)
+
+var metricsWorkerStartNames = []string{
+	metricsLabelWorkerName,
+}
+
+var metricsModelWorkerStartNames = []string{
+	metricsLabelModelUUID,
+}
+
+// Collector defines a new metrics collector. This allows the collection of
+// models for one model.
+type Collector struct {
+	workerStarts      *prometheus.GaugeVec
+	modelWorkerStarts *prometheus.GaugeVec
+}
+
+// NewMetrics creates a new collector for a model.
+func NewMetrics() *Collector {
+	return &Collector{
+		workerStarts: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Namespace: dependencyMetricsNamespace,
+			Subsystem: dependencySubsystemNamespace,
+			Name:      "worker_start",
+			Help:      "Current number of worker starts in the dependency engine by worker",
+		}, metricsWorkerStartNames),
+		modelWorkerStarts: prometheus.NewGaugeVec(prometheus.GaugeOpts{
+			Namespace: dependencyMetricsNamespace,
+			Subsystem: dependencySubsystemNamespace,
+			Name:      "worker_starts_for_model",
+			Help:      "Current number of worker starts in the dependency engine by model",
+		}, metricsModelWorkerStartNames),
+	}
+}
+
+// ForModel returns a metrics collector for a given model.
+func (c *Collector) ForModel(model names.ModelTag) dependency.Metrics {
```

Comment:
Are we actually using this anymore? Can we simplify the design if it is just a single metric that is using a model uuid?

Can we use a model name instead of UUID for human factors? Maybe setting both labels, but they have the same cardinality.

Reply:
Yes, I can add the model name though.